pwndbg/commands/mprotect.py

```diff
@@ -44,39 +45,37 @@ def prot_str_to_val(protstr):
 
 @pwndbg.commands.ArgparsedCommand(parser)
 @pwndbg.commands.OnlyWhenRunning
-@pwndbg.commands.OnlyAmd64
 def mprotect(addr, length, prot):
-    """Only x86_64."""
-    saved_rax = pwndbg.gdblib.regs.rax
-    saved_rbx = pwndbg.gdblib.regs.rbx
-    saved_rcx = pwndbg.gdblib.regs.rcx
-    saved_rdx = pwndbg.gdblib.regs.rdx
-    saved_rip = pwndbg.gdblib.regs.rip
 
     prot_int = prot_str_to_val(prot)
 
+    # generate a shellcode that executes the mprotect syscall
     shellcode_asm = pwnlib.shellcraft.syscall("SYS_mprotect", int(addr), int(length), int(prot_int))
```

Comment:
Hmm was there some reason why we don't set regs via `set $reg=val` and do it like this instead?

Reply:
(If we emitted only a single `syscall` we could just do a single `stepi` instead of `nextsyscall` + `stepi`)

Reply:
Not really, x86 (32bit) passes the arguments via the stack for example

Reply:
Yes and no: x86 calling convention does indeed pass arguments on the stack but arguments to syscalls are passed via registers; see https://chromium.googlesource.com/chromiumos/docs/+/master/constants/syscalls.md#x86-32_bit

Reply:
Unfortunately there is no such information in the architecture structs for the syscall number register, I will add it in another pull request and fix this, though.